woocommerce/woocommerce-ios
# Product Creation AI: Use eligibility checker in coordinator

Closes: #10689

## Description

Checks for AI eligibility in coordinator. 

Changes
- Check the feature flag from the eligibility checker
- Receive store has products info in `isEligible` method instead of `init` method. This is needed to allow dependency injection and unit testing
- Fix the missed storage property update in https://github.com/woocommerce/woocommerce-ios/pull/10710/commits/c02c2e6195d88ab0c45727c87f815bb55d17bf84
- Add/update unit tests

## Testing instructions
- Create a JN site with Jetpack AI plugin. (Checkbox available in JN home page to install this plugin)
- Tap the add new product task on store onboarding. 
- Validate that the new AI action sheet is displayed.
- Switch to Products tab, select the "+" button to add a new product. The same AI action sheet should be displayed.
- Switch to a JN site with no Jetpack AI plugin and validate that you don't see the AI action sheet.

## Screenshots
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/7298fbc3-1467-4cf9-8c50-3f671d5af592" width=320 />

---
